model/parameter.go

```diff
@@ -56,6 +56,15 @@ type ParameterDefinition struct {
 	RelevantIf     []RelevantIfCondition `json:"relevantIf,omitempty" yaml:"relevantIf,omitempty" mapstructure:"relevantIf"`
 	Hidden         bool                  `json:"hidden" yaml:"hidden"`
 	AdvancedConfig bool                  `json:"advancedConfig" yaml:"advancedConfig" mapstructure:"advancedConfig"`
+
+	Options ParameterOptions `json:"options" yaml:"options"`
+}
+
+// ParameterOptions specify further customization for inputs in the UI
```

Comment:
nit: Technically this will work anywhere, right? Not just the ui.

```suggestion
// ParameterOptions specify further customization for input parameters
```

Reply:
True.  This would change the validation for an `enum`